### PR TITLE
Adds interfaces to represent the behavior of kurmad setup

### PR DESCRIPTION
**Pull request by wallyqs**
Example testing of bootstrap sequence, formatting updates to some logging messages as well.
Added a couple of example tests for the bootstrap sequence in case we want to follow this idea...
@krobertson @jpoler

### CODE REVIEW

**Comment by krobertson** [comment]
So this allows testing the interaction of some of the setup functions? Since some depend on others, and it allows mocking out specific ones.

Agree too with some of the logging changes to be clearer about skipping some of the setup rather than an making it look like an error.

**Comment by wallyqs** [comment]
Yeah think we could potentially add more tests around the interactions between the funcs following something like this.

File: kurmad/runner.go [hunk 1]
```diff
@@ -23,8 +23,34 @@ import (
...
+	createDirectories() error
+	createImageManager() error
+	prefetchImages()
+	createPodManager() error
+	createNetworkManager()
+	startDaemon() error
+	startInitialPods()
+}
+
+// runner is an object that is used to handle the startup of the full KurmaOS
```
Comment (jpoler) [inline kurmad/runner.go]: I'm confused about this, is this KurmaOS or just Kurma?

File: kurmad/runner.go [hunk 2]
```diff
@@ -181,11 +208,11 @@ func (r *runner) createPodManager() error {
 
 // createNetworkManager creates the network manager which launches the network
 // provisioner pods.
-func (r *runner) createNetworkManager() error {
+func (r *runner) createNetworkManager() {
 	networkManager, err := networkmanager.New(r.podManager)
 	if err != nil {
-		r.log.Errorf("Failed to create network manager: %v", err)
-		return nil
+		r.log.Errorf("Skip networking since failed to create network manager: %v", err)
```
Comment (jpoler) [inline kurmad/runner.go]: maybe `"Skipping networking due because creation of network manager failed"`?

Comment (jpoler) [inline kurmad/runner.go]: This is more of an opinion question, but if the manifest specifies that it wants a network manager, and we fail here, should that be considered a hard failure?

File: kurmad/runner.go [hunk 3]
```diff
@@ -242,18 +269,19 @@ func (r *runner) startDaemon() error {
...
-func (r *runner) startInitialPods() error {
+func (r *runner) startInitialPods() {
 	for d, ip := range r.config.InitialPods {
 		name, podManifest, err := ip.Process(r.imageManager)
 		if name == "" {
 			name = fmt.Sprintf("initial-pod-%d", d+1)
 		}
 		if err != nil {
-			r.log.Errorf("Failed to configure pod %q: %v", name, err)
+			r.log.Errorf("Skip creating pod %q since failed to configure: %v", name, err)
```
Comment (jpoler) [inline kurmad/runner.go]: I suggest `Failed to configure pod %q, skipping`.

File: kurmad/runner_test.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
+// Copyright 2015 Apcera Inc. All rights reserved.
```
Comment (jpoler) [inline kurmad/runner_test.go]: 2016

File: kurmad/runner_test.go [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+func (r *dummyFailRunner) createImageManager() error { return nil }
+func (r *dummyFailRunner) prefetchImages()           {}
+func (r *dummyFailRunner) createPodManager() error {
+	return &dummyCreatePodManagerError{ErrorCode: 90}
+}
+func (r *dummyFailRunner) createNetworkManager() {}
+func (r *dummyFailRunner) startDaemon() error    { return nil }
+func (r *dummyFailRunner) startInitialPods()     {}
+
+func TestBootstrapFailure(t *testing.T) {
```
Comment (jpoler) [inline kurmad/runner_test.go]: I like.

File: kurmad/runner.go (hunk 3 above)
Comment (wallyqs) [inline kurmad/runner.go]: sounds good, updated

File: kurmad/runner.go (hunk 2 above)
Comment (wallyqs) [inline kurmad/runner.go]: Let's keep behavior as is in current PR, can update next if we want to change

File: kurmad/runner.go (hunk 1 above)
Comment (wallyqs) [inline kurmad/runner.go]: reworded

**Comment by jpoler** [comment]
LGTM

**Pull request merged**